src/ciphersuite/mod.rs

```diff
@@ -258,7 +267,9 @@ impl Ciphersuite {
     ) -> HpkeCiphertext {
         // TODO: put hpke in the ciphersuite.
         let hpke = Hpke::new(Mode::Base, self.hpke_kem, self.hpke_kdf, self.hpke_aead);
-        let (kem_output, ciphertext) = hpke.seal(&pk_r.value, info, aad, ptxt, None, None, None);
+        let (kem_output, ciphertext) = hpke
+            .seal(&pk_r.into(), info, aad, ptxt, None, None, None)
```

Comment:
Do I understand correctly, that the `pk_r` (or `sk_r` further below) is cloned here? I think you mentioned that we wanted to avoid cloning key material, so I assume that it's promptly consumed by `seal`?

Reply:
Yes, this clones right now. But same argument as below 🙂

Reply:
Yes, this clones. But same comment as below 🙂 So this should go away soon.